src/HttpCache/CacheTagsFormatterInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\HttpCache;
+
+/**
+ * Allows cache purgers to format the cache tags header according
+ * to their own needs.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+interface CacheTagsFormatterInterface
```

Comment:
What about just using `callable`. For this use case it's enough IMO, no need to introduce an interface.

Reply:
Callables are not a good choice at all here. They are very hard to add. Right now you can simply configure a different service by using the `tags_formatter` config setting and you're done.

Reply:
Formatting tags is very specific edge case to me, ok to allow to do it, but I'm not confortable to add so many things just for this. A `callable` looks perfect for this use case.

Reply:
I've always said it's something that's specific to the purger because the purger knows how to format the tags so it can actually purge them. It never makes sense to arbitrarily combine purgers and tag formatters. That's why I had `interface CacheTagsFormattingPurgerInterface extends PurgerInterface` and passed the purger to the `AddTagsListener`. That way, any purger can format if it needs but it can also just leave this interface alone and go with the default implementation which is just a CSV list. This is still the version that makes the most sense from a software architectural POV to me because responsibilities are put where they belong.